pandas/io/parsers.py

```diff
@@ -1013,6 +1016,26 @@ def tostr(x):
 
         return names, index_names, col_names, passed_names
 
+    def _maybe_dedup_names(self, names):
+        # see gh-7160 and gh-9424: this helps to provide
+        # immediate alleviation of the duplicate names
+        # issue and appears to be satisfactory to users,
+        # but ultimately, not needing to butcher the names
+        # would be nice!
+        if self.mangle_dupe_cols:
+            names = list(names)  # so we can index
+            counts = {}
+
+            for i, col in enumerate(names):
+                cur_count = counts.get(col, 0)
+
+                if cur_count > 0:
+                    names[i] = '%s.%d' % (col, cur_count)
+
+                counts[col] = cur_count + 1
```

Comment:
More pythonic?  Hmm...I figured unless you were going to use functionality in `defaultdict` that isn't in `dict`, using `defaultdict` is slight overkill.  But that is just my opinion.